xarray/core/groupby.py

```diff
@@ -473,7 +473,7 @@ def lookup_order(dimension):
             return axis
 
         new_order = sorted(stacked.dims, key=lookup_order)
-        return stacked.transpose(*new_order)
+        return stacked.transpose(*new_order, transpose_coords=True)
```

Comment:
Should this be `False`?

Reply:
We should probably do the same thing for this transpose in groupby that we did for transpose() in general -- we should have a keyword argument to control this behavior, e.g., perhaps `ds.groupby('something', restore_coord_dims=True).mean()`?

We definitely don't want to issue warnings internally that users can't control. `groupby()` using `transpose()` internally is an implementation detail that isn't part of xarray's public API.